src/ui/simulator/application/study.cpp

```diff
@@ -969,30 +973,16 @@ SaveResult ExportMap(const Yuni::String& path,
 
         // The binding constraints data must be reloaded since the current
         // code is not able to dynamically reload it by itself
-        study.ensureDataAreLoadedForAllBindingConstraints();
+        study->ensureDataAreLoadedForAllBindingConstraints();
         // Reload runtime info about the study (Paranoid, should not be required)
-        if (study.uiinfo)
-            study.uiinfo->reloadAll();
+        if (study->uiinfo)
+            study->uiinfo->reloadAll();
 
         // GUIFlagInvalidateAreas = true;
         OnStudySavedAs();
     }
 
-    Menu::AddRecentFile(mainFrm.menuRecentFiles(),
-                        wxStringFromUTF8(study.header.caption),
-                        wxStringFromUTF8(study.folder));
-
-    // Rebuild the menu
-    Menu::RebuildRecentFiles(mainFrm.menuRecentFiles());
-
-    gLastOpenedStudyFolder = wxStringFromUTF8(study.folder);
-
-    RefreshListOfOutputsForTheCurrentStudy();
-
-    mainFrm.refreshMenuInput();
-    mainFrm.refreshMenuOptions(&study);
-    mainFrm.refreshStudyLogs();
-
+    finalizeSaveExport(study, mainFrm);
```

Comment:
you introduce **finalizeSaveExport**, which is fine.
But there is a lot of duplicated code left in functions **SaveStudyAs** and **ExportMap** by the way.

Reply:
Right, but it's within a pair of brackets with `StudyUpdateLocker` called at the start. Hence it's harder to factorize this part.